src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -29,12 +29,18 @@ internal NpgsqlDatabaseInfo DatabaseInfo
         get => _databaseInfo ?? throw new InvalidOperationException("Internal error: this type mapper hasn't yet been bound to a database info object");
         set
         {
+            // We're attempting to set the same NpgsqlDatabaseInfo as we already have.
+            // If so, there is no reason to reset type mappings since NpgsqlDatabaseInfo is immutable.
+            // This might happen with multiplexing due to sharing the same ConnectorTypeMapper between multiple connections.
+            if (_databaseInfo == value)
```

Comment:
I tend to use ReferenceEquals in these cases to be extra sure it's a reference identity comparison